sklearn/metrics/tests/test_common.py

```diff
@@ -951,33 +974,36 @@ def check_sample_weight_invariance(name, metric, y1, y2):
 
     # check that unit weights gives the same score as no weight
     unweighted_score = metric(y1, y2, sample_weight=None)
-    assert_almost_equal(
+
+    assert_array_almost_equal(
         unweighted_score,
         metric(y1, y2, sample_weight=np.ones(shape=len(y1))),
         err_msg="For %s sample_weight=None is not equivalent to "
                 "sample_weight=ones" % name)
 
     # check that the weighted and unweighted scores are unequal
     weighted_score = metric(y1, y2, sample_weight=sample_weight)
-    assert_not_equal(
-        unweighted_score, weighted_score,
-        msg="Unweighted and weighted scores are unexpectedly "
-            "equal (%f) for %s" % (weighted_score, name))
+    assert_raises(
+        AssertionError,
+        assert_array_equal, unweighted_score, weighted_score,
```

Comment:
Use approximate equality here, please

Reply:
You mean `assert_array_almost_equal` right? If I do that it will make the test fail `micro_average_precision_score`

```
    def _raiseFailure(self, standardMsg):
        msg = self.test_case._formatMessage(self.msg, standardMsg)
>       raise self.test_case.failureException(msg)
E       AssertionError: Unweighted and weighted scores are unexpectedly almost equal (0.118667604216) and (0.118668873707) for micro_average_precision_score

msg        = 'Unweighted and weighted scores are unexpectedly almost equal (0.118667604216) and (0.118668873707) for micro_average_precision_score'
self       = <sklearn.utils._unittest_backport._AssertRaisesContext instance at 0x7ff47bf8f758>
standardMsg = 'AssertionError not raised'
```

Reply:
I think micro_average_precision_score should be identical to accuracy_score. Perhaps this is not as robust a test as we thought?  Does that still fail if assert_allclose is used? It relies on relative tolerance rather than absolute and is probably a better test here

Reply:
Interesting. The test passes when using `assert_allclose`. 

Additionally the use of `assert_array_almost_equal` is dis-advised by NumPy. I decided to replace all approximate equality checks with `assert_allclose` to increase overall robustness of the invariance tests.